src/clr-angular/data/tree-view/models/declarative-tree-node-model.spec.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { DeclarativeTreeNodeModel } from './declarative-tree-node-model';
+import { TreeNodeModel } from './tree-node-model';
+
+export default function(): void {
+  describe('DeclarativeTreeNodeModel', () => {
+    let parent: DeclarativeTreeNodeModel<void>;
+
+    beforeEach(function() {
+      parent = new DeclarativeTreeNodeModel(null);
+    });
+
+    afterEach(function() {
+      parent.destroy();
+    });
+
+    it('extends TreeNodeModel', function() {
+      expect(parent instanceof TreeNodeModel).toBeTrue();
+    });
+
+    it('registers to its parent on creation', function() {
+      const child1 = new DeclarativeTreeNodeModel<void>(parent);
+      const child2 = new DeclarativeTreeNodeModel<void>(parent);
+      expect(parent.children).toEqual([child1, child2]);
```

Comment:
You aren't calling `child1.destroy()` and `child2.destroy()` so memory leak?

Reply:
I was being overly cautious in the other places, but for this one the Observables are never subscribed, so there is no memory leak. I can still destroy them if you want.